vcd/metadata.go

```diff
@@ -4,9 +4,85 @@ import (
 	"fmt"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
 	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"regexp"
+	"strings"
 )
 
+// ignoreMetadataSchema returns the schema associated to ignore_metadata for the provider configuration.
+func ignoreMetadataSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeSet,
+		Optional:    true,
+		Description: "Defines a set of `metadata_entry` that need to be ignored by this provider. All filters on this attribute are computed with a logical AND",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"object_type": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Ignores metadata from the specific object type in VCD",
+				},
+				"object_name": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Ignores metadata from the specific object in VCD named like this argument",
+				},
+				"key_regex": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Regular expression of the metadata entry keys to ignore. Either `key_regex` or `value_regex` is required",
+					// Note: AtLeastOneOf can't be used here
```

Comment:
Could  `ExactlyOneOf` work here if this is not Terraform limitation that it doesn't work? (I see this is being done https://github.com/vmware/terraform-provider-vcd/pull/1057/files#diff-194d724e0d1ac9bf551517de0ed3d86746f2a11776b9e404b13054e31c156fdaR74)

Reply:
Could be, but in this case it would cause that you either populate `key_regex` or `value_regex`, if you populate both it would fail, right?
What I wanted is that one should at least use one of them.

Reply:
You're right, I had wrong assumption